src/webots/gui/WbStreamingServer.cpp

```diff
@@ -401,6 +411,16 @@ void WbStreamingServer::processTextMessage(QString message) {
         sendWorldStateToClient(client, state);
     }
     sendToClients("reset finished");
+  } else if (message == "revert")
+    WbApplication::instance()->worldReload();
+  else if (message.startsWith("load:")) {
+    WbWorld *world = WbWorld::instance();
+    const QString worldName = message.mid(5);
+    const QString fullPath = QFileInfo(world->fileName()).dir().absolutePath() + '/' + worldName;
```

Comment:
We should probably check that worldName contains a simple file name without any path component, e.g., to avoid things like `../../my_secret_project/worlds/my_secret_simulation.wbt`.